packages/internal-test-env/index.ts

```diff
@@ -87,69 +86,57 @@ export interface EnvVariables {
-  if (process.env.CI) {
-    return;
-  }
-
   if (envLoaded) {
     return;
   }
+  if (!process.env.CI) {
+    const envPath = join(process.cwd(), "e2e/env/.env.local");
 
-  const envPath = join(process.cwd(), "e2e/env/.env.local");
-
-  // Otherwise load dotenv configuration
-  config({
-    path: envPath,
-  });
+    // Otherwise load dotenv configuration
+    config({
+      path: envPath,
+    });
 
-  if (!process.env.E2E_TEST_ENVIRONMENT) {
-    console.error(
-      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
-tried looking in the following directory for \`.env.local \`: 
-${envPath}`
-    );
+    if (!process.env.E2E_TEST_ENVIRONMENT) {
+      console.error(
+        `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
+  tried looking in the following directory for \`.env.local \`: 
+  ${envPath}`
+      );
+    }
   }
-
-  // Creating feature flag object if there are feature flags
-  Object.keys(process.env)
-    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
-    .forEach((key) => (featuredFlags[key] = process.env[key]));
-  // Marking env as loaded
+  // If we're in CI, the environment is already configured, and we just loaded
+  // it otherwise.
   envLoaded = true;
 }
 
-function getTestingEnvironment(
-  environment: unknown
-): asserts environment is EnvVariables {
-  // Populate your process.env from your .env file of choice
-
-  // TODO: Replace these inline validations and checks with envalid or env-var
-  if (
-    !availableEnvironment.includes(
-      (environment as EnvVariables).E2E_TEST_ENVIRONMENT as AvailableEnvironment
-    )
-  ) {
-    throw new Error(
-      `Unknown environment: [${
-        (environment as EnvVariables).E2E_TEST_ENVIRONMENT
-      }]`
-    );
-  }
-
-  if (typeof (environment as EnvVariables).E2E_TEST_IDP !== "string") {
-    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
-  }
-}
-
 function getBaseTestingEnvironment<T extends LibraryVariables>(libVars?: T
   ): T extends NodeVariables ? TestingEnvironmentNode : TestingEnvironmentBrowser   {
     setupEnv();
-    getTestingEnvironment(process.env);
+    // TODO: Replace these inline validations and checks with envalid or env-var
+    
+    // Load and validate target environment name.
+    const targetEnvName = process.env.E2E_TEST_ENVIRONMENT;
+    if (
+      !availableEnvironment.includes(targetEnvName as AvailableEnvironment)
+    ) {
+      throw new Error(`Unknown environment: [${targetEnvName}]`);
+    }
+
+    // Load and validate target OpenID Provider.
+    const targetOp = process.env.E2E_TEST_IDP;
+    if (typeof targetOp !== "string") {
+      throw new Error("The environment variable E2E_TEST_IDP is undefined.");
+    }
+
+    // Creating feature flag object if there are feature flags
+    Object.keys(process.env)
+    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
+    .forEach((key) => (featuredFlags[key] = process.env[key]));
```

Comment:
I think prettier should suggest this here:

```suggestion
      .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
      .forEach((key) => (featuredFlags[key] = process.env[key]));
```

But this still isn't right, it should be:

```suggestion
      .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
      .forEach((key) => (featuredFlags[key.substring(17).toLowerCase()] = process.env[key]));
```

So `E2E_TEST_FEATURE_FOO_BAR` would become `features.foo_bar`

Also worth double checking if `featuredFlags` needs to be global still with this change, I don't think it does, I think it can be a `const` inside this method, so a `reduce` instead of `forEach`.

Or:

```typescript
      const featureFlags = Object.keys(process.env)
        .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
        .reduce((flags, envVar) => {
          const flag = envVar.substring(17).toLowerCase();
          flags[flag] = process.env[envVar];
          return flags;
        }, {});
```

Reply:
Good point, I wasn't super happy about the global variable so here goes: 9f8f10f1b8cf1c0c52b333146501a3521c1406f8, and c840967385486184a502ef832ab0be6f6dcc3bfe